rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/connectorwrapper/Worker.java

```diff
@@ -101,10 +107,43 @@ public Worker(ConnectConfig connectConfig,
         this.offsetManagementService = offsetManagementService;
         this.taskPositionCommitService = new TaskPositionCommitService(this);
         this.plugin = plugin;
+
+        DefaultMQProducer producer = new DefaultMQProducer();
+        producer.setNamesrvAddr(connectConfig.getNamesrvAddr());
+        producer.setInstanceName(ConnectUtil.createInstance(connectConfig.getNamesrvAddr()));
+        producer.setProducerGroup(connectConfig.getRmqProducerGroup());
+        producer.setSendMsgTimeout(connectConfig.getOperationTimeout());
+        producer.setMaxMessageSize(RuntimeConfigDefine.MAX_MESSAGE_SIZE);
+        producer.setLanguage(LanguageCode.JAVA);
+        this.producer = producer;
+
+        DefaultMQPullConsumer consumer = new DefaultMQPullConsumer();
+        consumer.setNamesrvAddr(connectConfig.getNamesrvAddr());
+        consumer.setInstanceName(ConnectUtil.createInstance(connectConfig.getNamesrvAddr()));
+        consumer.setConsumerGroup(connectConfig.getRmqConsumerGroup());
+        consumer.setMaxReconsumeTimes(connectConfig.getRmqMaxRedeliveryTimes());
+        consumer.setConsumerPullTimeoutMillis((long) connectConfig.getRmqMessageConsumeTimeout());
+        consumer.setLanguage(LanguageCode.JAVA);
+        this.consumer = consumer;
     }
 
     public void start() {
         taskPositionCommitService.start();
+
+        try {
+            this.producer.start();
+        } catch (MQClientException e) {
+            //todo print log
+            e.printStackTrace();
+        }
+
+        try {
+            this.consumer.start();
```

Comment:
It would be better to delay start the consumer/producer, just because of there's maybe have no SinkTask or SourceTask in a worker.

Reply:
ok, i solve it